test/utils/HookMiner.sol

```diff
@@ -0,0 +1,59 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+pragma solidity ^0.8.19;
+
+/// @title HookMiner - a library for mining hook addresses
+/// @dev This library is intended for `forge test` environments. There may be gotchas when using salts in `forge script` or `forge create`
+library HookMiner {
+    // mask to slice out the top 8 bit of the address
+    uint160 constant FLAG_MASK = 0xFF << 152;
+
+    // Maximum number of iterations to find a salt, avoid infinite loops
+    uint256 constant MAX_LOOP = 10_000;
+
+    /// @notice Find a salt that produces a hook address with the desired `flags`
+    /// @param deployer The address that will deploy the hook. In `forge test`, this will be the test contract `address(this)` or the pranking address
+    ///                 In `forge script`, this should be `0x4e59b44847b379578588920cA78FbF26c0B4956C` (CREATE2 Deployer Proxy)
```

Comment:
conventionally we don't have this kind of whitespace in our natspec, I have found some 4 space indentations for multiline natspec if you don't mind changing to that?